synapse/replication/tcp/streams/_base.py

```diff
@@ -200,6 +204,8 @@ def discard_updates_and_advance(self):
         if len(updates) >= MAX_EVENTS_BEHIND:
             raise Exception("stream %s has fallen behind" % (self.NAME))
 
+        # Due to the assertin above we know we're up to date, so we know that
+        # our new stream position is `current_token`.
```

Comment:
it's not an actual `assert`, so this is a bit misleading imho.

```suggestion
        # The update function didn't hit the limit, so we must have got all 
        # the updates to `current_token`, and can return that as our new
        # stream position.
```